features/step_definitions/common_steps.rb

```diff
@@ -268,22 +268,26 @@
   attach_file(label, File.join(Dir.pwd, 'fixtures', file))
 end
 
-When /^I click the top\-level summary table Edit link for the section '(.*)'$/ do |field_to_edit|
+When /^I click the top\-level summary table '(.*)' link for the section '(.*)'$/ do |link_name, field_to_edit|
   edit_link = page.find(:xpath, "//h2[contains(text(), '#{field_to_edit}')]/following-sibling::p[1]/a[text()]")
-  edit_link.text().should have_content('Edit')
+  edit_link.text().should have_content(link_name)
   edit_link.click
 end
 
-When /I click the summary table Edit link for '(.*)'$/ do |field_to_edit|
-  edit_link = page.find(:xpath, "//tr/*/span[contains(text(), '#{field_to_edit}')]/../..//a[text()]")
-  edit_link.text().should have_content('Edit')
+When /I click the summary table '(.*)' (link|button) for '(.*)'$/ do |link_name, elem_type, field_to_edit|
+  case elem_type
+    when 'link'
+      edit_link = page.find(:xpath, "//tr/*/span[text() = '#{field_to_edit}']/../..//a[contains(text(), '#{link_name}')]")
```

Comment:
Does `text()` trim surrounding whitespace?  Off the top of my head I don't think it does, so I worry that moving from `contains` to equal that we're not going to be able to match properly if the `<span>` isn't closely packed around the text we're searching for.

Reply:
Let me check

Reply:
Can indeed, Done 👍